error: conflicting types for ‘gettimeofday’

Build fails on Arch Linux:
```
/home/user/build/src/preeny/src/detime.c:24:5: error: conflicting types for ‘gettimeofday’
   24 | int gettimeofday(struct timeval *tv, struct timezone *tz)
      |     ^~~~~~~~~~~~
In file included from /home/user/build/src/preeny/src/detime.c:6:
/usr/include/sys/time.h:66:12: note: previous declaration of ‘gettimeofday’ was here
   66 | extern int gettimeofday (struct timeval *__restrict __tv,
      |            ^~~~~~~~~~~~
make[2]: *** [CMakeFiles/detime.dir/build.make:83: CMakeFiles/detime.dir/src/detime.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:688: CMakeFiles/detime.dir/all] Error 2
make: *** [Makefile:104: all] Error 2
==> ERROR: A failure occurred in build().
```

Thanks in advance!

--- a/src/detime.c
+++ b/src/detime.c
@@ -4,6 +4,7 @@
 #include <time.h>
 #ifdef __unix__
 #include <sys/time.h>
+#include <features.h>
 #endif
 
 #include "logging.h"
@@ -21,7 +22,11 @@ time_t time(time_t *res)
 }
 
 #ifdef __unix__
+#if !defined(__GLIBC__) || __GLIBC_PREREQ(2, 31)
+int gettimeofday(struct timeval *tv, void *__restrict tz)
+#else
 int gettimeofday(struct timeval *tv, struct timezone *tz)
+#endif
 {
 	
 	char *sec_str = getenv("TV_SEC");
@@ -32,6 +37,8 @@ int gettimeofday(struct timeval *tv, struct timezone *tz)
 	preeny_debug("gettimeofday frozen at %ld.%d\n", tv->tv_sec, tv->tv_usec);
 	
 	
+#if !defined(__GLIBC__) || __GLIBC_PREREQ(2, 31)
+#else
 	if (tz != NULL) {
 		char *minwest_str = getenv("TZ_MINWEST");
 		char *dst_str = getenv("TZ_DSTTIME");
@@ -40,6 +47,7 @@ int gettimeofday(struct timeval *tv, struct timezone *tz)
 
 		preeny_debug("gettimeofday tz frozen at -%d minm, DST: %d\n", tz->tz_minuteswest, tz->tz_dsttime);
 	}
+#endif
 
 	return 0;
 }

